project_task_code/models/project_task.py

```diff
@@ -42,3 +42,15 @@ def name_get(self):
             name = "[{}] {}".format(rec.code, task[1])
             new_result.append((rec.id, name))
         return new_result
+
+    @api.model
+    def _name_search(
+        self, name, args=None, operator="ilike", limit=100, name_get_uid=None
+    ):
+        if not args:
+            args = []
+        if name:
+            args += ["|", ["code", "ilike", name]]
```

Comment:
**issue:** only add new domain if operator is ilike.

Otherwise you could be getting to weird domains when call uses a different operator.